src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/EJSONWrapper.java

```diff
@@ -118,4 +121,46 @@ private static byte[] bytesFromByteBuffer(ByteBuffer buffer) {
     buffer.get(bytes);
     return bytes;
   }
+
+  // Returns the float array value from the binary EJSON wrapper
+  public float[] getVectorValueForBinary() {
+    if (type != EJSONType.BINARY) {
+      throw new RuntimeException("Vector value can only be extracted from binary EJSON wrapper");
+    }
+    return toFloatArray(value.textValue());
+  }
+
+  // Utility method for converting binary wrapped value to float array
+  public static float[] toFloatArray(String base64Vector) {
```

Comment:
This looks similar / duplicate to JsonCodec.byteArrayFromEJSON() ok to merge but can you make sure to discuss with @tatu to see if there is a better way to centralise this. 

Also, it would be prob be best to use IllegalArgumentException rather than change to RuntimeException, good to give a better message with more context, and also pass the original error as the cause when building the new error